### PR TITLE
Add support for tags

### PR DESCRIPTION
**Pull request by justenwalker**
For #10 

Add support for `tags` in the service configuration. Didn't seem like too much work to add, so I'm afraid I may have glossed over some important details. 

Another set of eyes appreciated.

### CODE REVIEW

File: src/containerbuddy/config.go [hunk 1]
```diff
@@ -65,6 +65,7 @@ type ServiceConfig struct {
 	Port             int             `json:"port"`
 	TTL              int             `json:"ttl"`
 	Interfaces       json.RawMessage `json:"interfaces"`
+	Tags             []string        `json:"tags,omitempty"`
```
Comment (tgross) [inline src/containerbuddy/config.go]: I think we're going to need to include tags for the `BackendConfig` as well. It's my understanding that the primary use case for this that @misterbisson was talking about in #10 was being able to split out services for e.g. dev vs prod. So when we query Consul we're probably going to want to have a list of tags to send to [`checkHealth`](https://github.com/joyent/containerbuddy/blob/7249deb6e69fd66469a3f4ed366186d28937769b/src/containerbuddy/consul.go#L84) and then on to [`Health.Service`](https://godoc.org/github.com/hashicorp/consul/api#Health.Service)

**Comment by justenwalker** [comment]
@tgross What would the filter behavior be for tags on a Backend. Would they filter with AND (Must have every tag) or with OR (must have one of the tags)

**Comment by justenwalker** [comment]
I checked the [consul codebase](https://github.com/hashicorp/consul/blob/fed14582145987a6f56f4e21c841f9bfd97d411f/consul/state/state_store.go) for clarification.  It just accepts 1 tag (ignoring case) and filters if that tag is found.  So I think we should just use a single tag.

**Comment by tgross** [comment]
I'm feeling like there's going to come a point where we want query-with-multiple-tags support but without a backend that supports it I feel that'd be YAGNI. It's easy enough to add it in later.

File: src/containerbuddy/config_test.go [hunk 2]
```diff
@@ -85,6 +87,15 @@ func TestValidConfigParse(t *testing.T) {
 		t.Errorf("Expected 3 args but got unexpected results: %v", args)
 	}
 
+	expectedTags := []string{"tag1", "tag2"}
```
Comment (tgross) [inline src/containerbuddy/config_test.go]: Can we make sure we have a test for empty tags too? Just to verify that we don't regress with it down the road?

**Comment by tgross** [comment]
Ok, this LGTM.

**Pull request merged**